### PR TITLE
Publish devfiles with registryUrls to GH pages

### PR DESCRIPTION
**Pull request by ericwill**
Signed-off-by: Eric Williams <ericwill@redhat.com>

### What does this PR do?
Generates devfiles with `registryUrl` fields and publishes them on GitHub pages. For example, the URL of the `7.29.1` devfile registry hosted on GitHub pages will be `eclipse-che.github.io/che-devfile-registry/7.29.1/`

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

Fixes eclipse/che#19269

### How to test this PR?

Run the script with your GitHub token and `VERSION` set to a valid tag, and observe the `gh-pages` branch of this repo.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: .ci/generate_and_publish_registry_url_devfiles.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+done
+
+git config --global user.email "che-bot@eclipse.org"
+git config --global user.name "CHE Bot"
+
+# Make temporary directory and copy out devfiles and images
+mkdir -p /tmp/content/"${VERSION}"
+./build.sh --tag gh-pages-generated
+podman rm -f devfileRegistry
+podman create --name devfileRegistry quay.io/eclipse/che-devfile-registry:gh-pages-generated
```
[the remark this answers is not in the document]
Comment (ericwill) [reply .ci/generate_and_publish_registry_url_devfiles.sh:48]: Oops, should be docker (was using podman for testing)

[the remark this answers is not in the document]
Comment (benoitf) [reply .ci/generate_and_publish_registry_url_devfiles.sh:48]: you can use ${BUILDER}

[the remark this answers is not in the document]
Comment (sunix) [reply .ci/generate_and_publish_registry_url_devfiles.sh]: I would make them into 2 separate scripts:
1. to extract with podman
2. clone and push to gh-pages

so 2. could be reused in elsewhere (for instance a dogfooding workspace)

[the remark this answers is not in the document]
Comment (ericwill) [reply .ci/generate_and_publish_registry_url_devfiles.sh]: > I would make them into 2 separate scripts:
> 
>     1. to extract with podman
> 
>     2. clone and push to gh-pages
> 
> 
> so 2. could be reused in elsewhere (for instance a dogfooding workspace)

As discussed let's do it in a later PR. This one is meant specifically for the release flow.

**Pull request merged**